backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java

```diff
@@ -69,43 +70,147 @@ public static BlobDiff setConflictBlobsInDiff(
+      AutogenBlobDiff diff0 = AutogenBlobDiff.fromProto(blobDiff);
+      AutogenBlobDiff diff1 = AutogenBlobDiff.fromProto(blobDiff);
+      switch (locSpecificBlobDiffA.get(0).getContentCase()) {
+        case CODE:
+          List<AutogenCodeDiff> diffs = getCodeConflictBlob(a.getCode(), b.getCode(), c.getCode());
+          if (diffs.size() >= 1) {
+            return Arrays.asList(
+                diff0.setCode(diffs.get(0)).toProto().build(),
+                diff1.setCode(diffs.get(1)).toProto().build());
+          } else {
+            return Collections.singletonList(diff0.setCode(diffs.get(0)).toProto().build());
+          }
+        case CONFIG:
+          return Collections.singletonList(
+              diff0
+                  .setConfig(getConfigConflictBlob(a.getConfig(), b.getConfig(), c.getConfig()))
+                  .toProto()
+                  .build());
+        case DATASET:
+          List<AutogenDatasetDiff> datasetDiffs =
+              getDatasetConflictBlob(a.getDataset(), b.getDataset(), c.getDataset());
+          if (datasetDiffs.size() >= 1) {
+            return Arrays.asList(
+                diff0.setDataset(datasetDiffs.get(0)).toProto().build(),
+                diff1.setDataset(datasetDiffs.get(1)).toProto().build());
+          } else {
+            return Collections.singletonList(
+                diff0.setDataset(datasetDiffs.get(0)).toProto().build());
+          }
+        case ENVIRONMENT:
+          List<AutogenEnvironmentDiff> envDiffs =
+              getEnvironmentonflictBlob(a.getEnvironment(), b.getEnvironment(), c.getEnvironment());
+          if (envDiffs.size() >= 1) {
+            return Arrays.asList(
+                diff0.setEnvironment(envDiffs.get(0)).toProto().build(),
+                diff1.setEnvironment(envDiffs.get(1)).toProto().build());
+          } else {
+            return Collections.singletonList(
+                diff0.setEnvironment(envDiffs.get(0)).toProto().build());
+          }
+        case CONTENT_NOT_SET:
+          return Collections.emptyList();
+      }
+    } else {
+      AutogenBlobDiff diff1 = AutogenBlobDiff.fromProto(blobDiff);
+      AutogenBlobDiff diff2 = AutogenBlobDiff.fromProto(blobDiff);
+      switch (locSpecificBlobDiffA.get(0).getContentCase()) {
+        case CODE:
+          diff1.setCode(getCodeConflictBlob(a.getCode(), b.getCode(), c.getCode()).get(0));
+          break;
+        case CONFIG:
+          diff1.setConfig(getConfigConflictBlob(a.getConfig(), b.getConfig(), c.getConfig()));
+          break;
+        case DATASET:
+          diff1.setDataset(
+              getDatasetConflictBlob(a.getDataset(), b.getDataset(), c.getDataset()).get(0));
+          break;
+        case ENVIRONMENT:
+          diff1.setEnvironment(
+              getEnvironmentonflictBlob(a.getEnvironment(), b.getEnvironment(), c.getEnvironment())
+                  .get(0));
+          break;
+        case CONTENT_NOT_SET:
+          break;
```

Comment:
Should we extract this and below comment in the separate function to reduce complexity? I think these two cases look the same.

Reply:
done